src/shared/gpt.c

```diff
@@ -222,26 +228,69 @@ const char *gpt_partition_type_uuid_to_string_harder(
         return sd_id128_to_uuid_string(id, buffer);
 }
 
-int gpt_partition_type_from_string(const char *s, GptPartitionType *ret) {
+int gpt_partition_type_from_string(const char *s, Architecture arch, GptPartitionType *ret) {
```

Comment:
please do not change the prototype, i.e. should be a xyz_full() thing instead, with a static inline wrapper to maintain the old prototype.

(but quite frankly, let's just do the translation stuff I suggested, nicer, simpler, more powerful)